src/ranged.cpp

```diff
@@ -1416,20 +1421,28 @@ static int rand_or_max( bool random, int max )
 // utility functions for projectile_attack
 double player::get_weapon_dispersion( const item *weapon, bool random ) const
 {
-    if( weapon->is_gun() && weapon->is_in_auxiliary_mode() ) {
-        const auto gunmod = weapon->gunmod_current();
-        if( gunmod != nullptr ) {
-            return get_weapon_dispersion( gunmod, random );
-        }
-    }
-
     double dispersion = 0.; // Measured in quarter-degrees.
     dispersion += skill_dispersion( *weapon, random );
 
+    if( is_driving() ) {
+        // get volume of gun (or for auxiliary gunmods the parent gun)
+        const item *parent = has_item( *weapon ) ? find_parent( *weapon ) : nullptr;
+        int vol = parent ? parent->volume() : weapon->volume();
+
+        ///\EFFECT_DRIVING reduces the inaccuracy penalty when using guns whilst driving
+        dispersion += std::max( vol - get_skill_level( skill_driving ), 1 ) * 20;
+    }
+
     dispersion += rand_or_max( random, ranged_dex_mod() );
     dispersion += rand_or_max( random, ranged_per_mod() );
 
-    dispersion += rand_or_max( random, 3 * (encumb(bp_arm_l) + encumb(bp_arm_r)));
+    // pistols select the least encumbered hand
+    if( weapon->gun_skill() == skill_pistol ) {
+        dispersion += rand_or_max( random, 3 * std::min( encumb( bp_arm_l ), encumb( bp_arm_r ) ) );
```

Comment:
That shouldn't halve the encumbrance when both arms are equally encumbered.

Also, it shouldn't be hardcoded to target the pistol skill, though this one can wait.